components/admins/components/validatorList.tsx

```diff
@@ -37,164 +51,154 @@
     }
   }, [modalId]);
 
+  const filteredValidators = useMemo(() => {
+    const validators = active ? activeValidators : pendingValidators;
+    return validators.filter(validator =>
+      validator.description.moniker.toLowerCase().includes(searchTerm.toLowerCase())
+    );
+  }, [active, activeValidators, pendingValidators, searchTerm]);
+
   const handleRemove = (validator: ExtendedValidatorSDKType) => {
     setValidatorToRemove(validator);
-    setWarningVisible(true);
+
     const modal = document.getElementById(`warning-modal`) as HTMLDialogElement;
     modal?.showModal();
   };
 
-  const filteredValidators = active
-    ? (Array.isArray(activeValidators) ? activeValidators : []).filter(
-        (validator) =>
-          validator.description.moniker
-            .toLowerCase()
-            .includes(searchTerm.toLowerCase()),
-      )
-    : (Array.isArray(pendingValidators) ? pendingValidators : []).filter(
-        (validator) =>
-          validator.description.moniker
-            .toLowerCase()
-            .includes(searchTerm.toLowerCase()),
-      );
-
   const [modalKey, setModalKey] = useState(0);
 
-  useEffect(() => {
-    if (modalId) {
-      const modal = document.getElementById(modalId) as HTMLDialogElement;
-      modal?.showModal();
-    }
-  }, [modalId, modalKey]);
-
   const handleRowClick = (validator: ExtendedValidatorSDKType) => {
     setSelectedValidator(validator);
-    setModalKey((prevKey) => prevKey + 1);
+    setModalKey(prevKey => prevKey + 1);
     setModalId(`validator-modal-${validator.operator_address}-${Date.now()}`);
   };
 
   return (
```

Comment:
This would only happen if the modal 'warning-modal' is not mounted which is a larger issue that will cause errors prior to reaching this possibility.